### DISCUSSION TITLE
supporting anthropic MCP servers

### DISCUSSION THREAD

**Discussion by rawwerks** [BerriAI/litellm#7059] [Ideas]
any interest from the berriAI team on modifying the litellm client to also serve as an anthropic "MCP Client", such that MCP servers can be called?

https://github.com/modelcontextprotocol

**Comment by runekaagaard** [comment]
Yeah I had the exact same idea. I'm using various chatgpt specific tools in emacs with litellm to use different LLM providers like Claude and having litellm handle the MCP part sounds like a great idea to enhance a LOT of different platforms with MCP. I would be open to work on it if it makes sense. There is a new quickstart repo with a client example here:

https://github.com/modelcontextprotocol/quickstart-resources/blob/main/mcp-client/client.py

I'm not sure if it's a good fit, what would talk against doing it?

**Comment by ghost** [comment]
hey @rawwerks @runekaagaard this is interesting. i can't find an api reference - am i misunderstanding mcp or is it just missing docs for how to call their api's? 

if someone has an e2e example here it would be great

-litellm maintainer

**Reply by runekaagaard** [reply]
The MCP servers are running locally and the client (in this case litellm would act as the client) communicates with them and ensures tools, resources and prompts are called when the LLM wants to do it. It offers the LLM multiple tries of calling tools, etc. before returning the prompt to the user.

I've tried several of the standalone clients on the list here: https://github.com/punkpeye/awesome-mcp-servers?tab=readme-ov-file#clients

However none of them (on my system, for me) are ready as a daily driver just yet.

The official docs for writing MCP clients can be found here: https://github.com/modelcontextprotocol/python-sdk?tab=readme-ov-file#writing-mcp-clients but it's a bit thin IMO.
